silkworm/rpc/http/connection.cpp

```diff
@@ -310,33 +341,10 @@ Task<void> Connection::do_write(const std::string& content, boost::beast::http::
 }
 
 void Connection::compress_data(const std::string& clear_data, std::string& compressed_data) {
-    z_stream strm;
-
-    std::memset(&strm, 0, sizeof(strm));
-    int ret = Z_OK;
-    ret = deflateInit2(&strm, Z_DEFAULT_COMPRESSION, Z_DEFLATED, 15 | 16, 8, Z_DEFAULT_STRATEGY);
-    if (ret != Z_OK) {
-        throw std::runtime_error("deflateInit2 fail");
-    }
-    strm.avail_in = static_cast<unsigned int>(clear_data.size());
-    strm.next_in = reinterpret_cast<const Bytef*>(clear_data.c_str());
-
-    do {
-        strm.next_out = reinterpret_cast<Bytef*>(temp_compressed_buffer_);
-        strm.avail_out = sizeof(temp_compressed_buffer_);
-
-        ret = deflate(&strm, Z_FINISH);
-        if (ret < 0) {
-            deflateEnd(&strm);
-            throw std::runtime_error("deflate fail");
-        }
-        if (compressed_data.size() < strm.total_out) {
-            // append the block to the output string
-            compressed_data.append(temp_compressed_buffer_, strm.total_out - compressed_data.size());
-        }
-    } while (ret != Z_STREAM_END);
-
-    deflateEnd(&strm);
+    boost::iostreams::filtering_ostream out;
+    out.push(boost::iostreams::gzip_compressor());
+    out.push(boost::iostreams::back_inserter(compressed_data));
+    boost::iostreams::copy(boost::make_iterator_range(clear_data), out);
```

Comment:
You can probably improve here by keeping the `boost::iostreams::filtering_ostream` instance and the compressed data buffer as state variables in `http::Connection` in order to have gzip initialisation/finalisation happen only once per connection (as opposed to doing it for every data compression). Next PR